container/layer_tools.bzl

```diff
@@ -36,8 +36,10 @@ def _extract_layers(ctx, artifact):
       "legacy": artifact,
   }
 
-def get_from_target(ctx, attr_target, file_target):
-  if hasattr(attr_target, "container_parts"):
+def get_from_target(ctx, attr_target, file_target, overriding_file_target=None):
+  if overriding_file_target:
+    return _extract_layers(ctx, overriding_file_target)
+  elif hasattr(attr_target, "container_parts"):
```

Comment:
I find this part of the change awkward / hard to reason about.

It is already awkward because of Bazel's `ctx.attr` vs. `ctx.file` split.

Why isn't the value passed to `overriding_file_target` simply passed to `file_target` instead?  What can we do to improve this interface?

Reply:
ctx.attr.base.files.to_list() seems to give the same info as ctx.files.base. I refactored the function a little bit, and now the single file_target passed to the function will override the rest. Let me know what do you think.